### PR TITLE
Add support for resources in javadoc jars and make variable substitution in javadoc arguments

### PR DESCRIPTION
**Pull request by vinnybod**
Note: This is a dupe of https://github.com/bazelbuild/rules_jvm_external/pull/1182 . I had to reopen the PR from a different GH org.

---

There's two use cases I'm trying to account for here.

1. I need to add resources to the javadoc jar such as a `LICENSE` file.
2. I need to send a `--define` variable to the javadoc generator so that the `maven_version` variable I have set will appear in the `<title>` of the generated html. rules_jvm_external already supports the same type of substitution on the `coordinates` attribute.

```
java_export(
    name = "lib",
    maven_coordinates="com.example:lib:$(maven_version)",
    ...
    javadocopts=["-windowtitle", "$(maven_version)"],
    doc_resources = ["//:LICENSE"]
)
```

### CODE REVIEW

File: private/rules/javadoc.bzl [hunk 1]
```diff
@@ -33,15 +34,27 @@ def generate_javadoc(
...
         args.add("-linkoffline")
         args.add(dep_info.url)
         args.add(dep_info.element_list.dirname)
         inputs.append(dep_info.element_list)
-    args.add_all(javadocopts)
+
+    processed_opts = []
+    for opt in javadocopts:
+        if "$(" in opt:
+            processed_opt = ctx.expand_make_variables("javadocopts", opt, ctx.var)
```
Comment (shs96c) [inline private/rules/javadoc.bzl:52]: Is there any harm in just expanding the make variables unconditionally? In the common case, I'd expect it to be a no-op, and it makes the code a little easier to read.

File: tests/com/github/bazelbuild/rules_jvm_external/BUILD [hunk 2]
```diff
@@ -105,3 +105,14 @@ java_test(
         artifact("org.hamcrest:hamcrest"),
     ],
 )
+
+java_library(
+    name = "ZipUtils",
```
Comment (shs96c) [inline tests/com/github/bazelbuild/rules_jvm_external/BUILD:110]: `zip_utils`. We save CamelCase for `java_binary` targets.

**Review by shs96c — CHANGES_REQUESTED** [review]
Two small nits, and then I'll merge. Thank you for re-opening!

File: private/rules/javadoc.bzl (hunk 1 above)
Comment (vinnybod) [reply private/rules/javadoc.bzl]: Seems to work!

**Comment by vinnybod** [comment]
@shs96c Made the suggested changes, this is ready for another review. Thanks!

**Review by shs96c — APPROVED** [review]
LGTM. Thank you for making the changes!

**Pull request merged**